src/Packages/Passport/Runtime/Scripts/Private/Model/Token.cs

```diff
@@ -0,0 +1,36 @@
+namespace Immutable.Passport.Model
+{
+    public class Token
+    {
+        public string Type { get; }
+
+        public string Quantity { get; }
+
+        public string? TokenId { get; }
+
+        public string? TokenAddress { get; }
+
+        private Token(string type, int quantity, string? tokenId = null, string? tokenAddress = null)
+        {
+            this.Type = type;
+            this.Quantity = $"{quantity}";
+            this.TokenId = tokenId;
+            this.TokenAddress = tokenAddress;
+        }
+
+        public static Token ETH(int quantity)
+        {
+            return new Token("ETH", quantity);
```

Comment:
Can we create enums for token types (ETH, ERC20, ERC721)? So that future functions can use it too?

Reply:
we can, but enums in C# are numeric only. I initially thought of using enums, and maybe extension methods, but it was a lot of converting integer to string and such, I didnt think it was the best option. Also, the TS SDK does not use enums because JS has a similar limitation.